packages/oscal-react-library/src/components/OSCALLoader.tsx

```diff
@@ -54,37 +64,68 @@ export function getRequestedUrl() {
   return new URLSearchParams(window.location.search).get("url");
 }
 
-export default function OSCALLoader(props) {
+type Renderer = (props: OSCALLoaderRendererProps) => ReactElement;
+
+type OscalWithSource = Oscal & { oscalSource?: string };
+
+interface OSCALDocumentLoaderProps {
+  renderForm: boolean;
+  backendUrl: string;
+  urlFragment: string;
+  isRestMode: boolean;
+}
+
+interface OSCALLoaderProps extends AnchorLinkProps {
+  isRestMode?: boolean;
+  hasDefaultUrl?: boolean;
+  backendUrl?: string;
+  oscalObjectType: any;
+  renderForm?: boolean;
+  renderer: Renderer;
+}
+
+interface OSCALLoaderRendererProps {
+  isRestMode: boolean | undefined;
+  oscalData: OscalWithSource;
+  oscalUrl: string;
+  onResolutionComplete: () => void;
+  handleRestSuccess: () => void;
+  handleRestError: (e: Error) => void;
+  handleFieldSave: (...args: any[]) => any;
+}
+
+export default function OSCALLoader(props: OSCALLoaderProps): ReactElement {
   const [isLoaded, setIsLoaded] = useState(false);
   const [isResolutionComplete, setIsResolutionComplete] = useState(false);
   const [hasDefaultUrl, setHasDefaultUrl] = useState(props.hasDefaultUrl);
-  const [oscalData, setOscalData] = useState([]);
+  const [oscalData, setOscalData] = useState(null as unknown as OscalWithSource);
```

Comment:
Just for my understanding, this is unideal right?

Reply:
Definitely -- and a holdover from before I defined `OscalWithSource`. `{} as OscalWithSource` is much better and what it is defined as now.